app/download/views.py

```diff
@@ -114,6 +114,17 @@ def get_document_metadata(service_id, document_id):
 @download_blueprint.route("/services/<uuid:service_id>/documents/<uuid:document_id>/authenticate", methods=["POST"])
 def authenticate_access_to_document(service_id, document_id):
     key = request.json.get("key")
+    if redis_store.exceeded_rate_limit(
+        f"authenticate-document-{service_id}-{document_id}",
+        limit=current_app.config["DOCUMENT_AUTHENTICATION_RATE_LIMIT"],
+        interval=current_app.config["DOCUMENT_AUTHENTICATE_RATE_INTERVAL"],
+    ):
+        return (
+            jsonify(error="Too many requests"),
```

Comment:
Optional: We could consider a more verbose error message like the API (although in this case it is just us as the users rather than service teams)

https://github.com/alphagov/notifications-api/blob/a81f34f8bd6a3fa44d358563464ae545e53ce758/app/v2/errors.py#L23